File: client/common/model/sku/sku-task-item.ts
import { LabelValue } from '@model/base/base'

export class SkuTaskItem {
  constructor(
    public id?: number,
    public taskId?: number,
    public url?: string,
    public status?: string,
    public source?: string,
    public remark?: string,
    public skuId?: number,
    public sourceSkuId?: string,
    public title?: string,
    public name?: string,
  ) {
    this.id = id
    this.taskId = taskId
    this.url = url
    this.status = status
    this.source = source
    this.remark = remark
    this.skuId = skuId
    this.sourceSkuId = sourceSkuId
    this.title = title
    this.name = name
  }
}

export enum SkuTaskItemStatus {
  PENDING = 'pending',
  SUCCESS = 'success',
  FAILED = 'failed',
  CANCEL = 'cancel',
  EXISTENCE = 'existence'
}

export class AddSkuTaskItemReq {
  constructor(
    public taskId: number,
    public url: string,
    public status: string,
    public source: string,
    public remark?: string,
    public skuId?: number,
    public id?: number,
    public sourceSkuId?: string,
    public title?: string,
  ) {
    this.taskId = taskId
    this.url = url
    this.status = status
    this.source = source
    this.remark = remark
    this.skuId = skuId
    this.id = id
    this.sourceSkuId = sourceSkuId
    this.title = title
  }
}
export class SkuTaskItemResp {
  constructor(
    public id: number,
    public taskId: number,
    public url: string,
    public status: string,
    public remark: string,
    public skuId: number,
    public name: string,
    public sourceSkuId: string,
    public source: string,
    public sourceLableValue: LabelValue,
    public createdAt: string,
    public newSkuUrl: string,
    public title: string
  ) {
    this.id = id
    this.taskId = taskId
    this.url = url
    this.status = status
    this.remark = remark
    this.skuId = skuId
    this.name = name
    this.sourceSkuId = sourceSkuId
    this.source = source
    this.sourceLableValue = sourceLableValue
    this.createdAt = createdAt
    this.newSkuUrl = newSkuUrl
    this.title = title
  }
}

File: server/app/sku/vo/sku_task_item.go
package vo

import (
	"server/common/base"
	"server/common/base/vo"
)

type GetSkuTaskItemListReq struct {
	TaskID uint64 `uri:"id"`
}

type GetSkuTaskItemListResp struct {
	ID               uint64          `json:"id"`
	TaskId           uint64          `json:"taskId"`
	Url              string          `json:"url"`
	Status           string          `json:"status"`
	StatusLableValue vo.LabelValueVO `json:"statusLableValue"`
	Remark           string          `json:"remark"`
	SkuID            uint64          `json:"skuId"`
	Name             string          `json:"name"`
	SourceSkuID      string          `json:"sourceSkuId"`
	Source           string          `json:"source"`
	SourceLableValue vo.LabelValueVO `json:"sourceLableValue"`
	Title            string          `json:"title"`
	CreatedAt        base.Time       `json:"createdAt"`
	NewSkuUrl        string          `json:"newSkuUrl"`
}

type AddSkuTaskItemReq struct {
	ID          uint64 `json:"id"`
	TaskID      uint64 `json:"taskId"`
	Url         string `json:"url" binding:"required"`
	Status      string `json:"status" binding:"required"`
	Remark      string `json:"remark"`
	Source      string `json:"source"`
	SkuID       uint64 `json:"skuId"`
	SourceSkuId string `json:"sourceSkuId"`
	Title       string `json:"title"`
}
